tools/e2e-agent/disk.go

```diff
@@ -0,0 +1,190 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+const (
+	// DiskImageSize is the default file size(1GB) used while creating backing image
+	DiskImageNamePrefix = "openebs-disk"
+)
+
+// LoopDevice has the attributes of a virtual disk which is emulated for testing
+type LoopDevice struct {
+	// Size in bytes
+	Size int64 `json:"size"`
+	// The backing image name
+	// eg: fake123
+	ImageName string `json:"imageName"`
+	// Image directory
+	// eg: /tmp
+	ImgDir string `json:"imgDir"`
+	// the disk name
+	// eg: /tmp/loop9002
+	DiskPath string `json:"diskPath"`
+}
+
+func (disk *LoopDevice) createDiskImage() error {
+	f, err := os.CreateTemp(disk.ImgDir, DiskImageNamePrefix+"-*.img")
+	if err != nil {
+		return fmt.Errorf("error creating disk image. Error : %v", err)
+	}
+	disk.ImageName = f.Name()
+	err = f.Truncate(disk.Size)
+	if err != nil {
+		return fmt.Errorf("error truncating disk image. Error : %v", err)
+	}
+
+	return nil
+}
+
+// Generates a random loop device name. The name will be of the
+// format loop9XXX, where X=[0-9]. The 4 digit numbering is chosen so that
+// we get enough disks to be randomly generated and also it does not clash
+// with the existing loop devices present in some systems.
+func getLoopDevName() string {
+	rand.Seed(time.Now().UTC().UnixNano())
+	randomNumber := 9000 + rand.Intn(999)
+	diskName := "loop" + strconv.Itoa(randomNumber)
+	return diskName
+}
+
+// CreateLoopDevice creates a loop device if the disk is not present
+func CreateLoopDevice(w http.ResponseWriter, r *http.Request) {
+	var device LoopDevice
+	var msg string
+
+	d := json.NewDecoder(r.Body)
+	if err := d.Decode(&device); err != nil {
+		msg = fmt.Sprintf("failed to read JSON encoded data, Error: %s", err.Error())
+		klog.Error(msg)
+		WrapResult(msg, ErrJsonDecode, w)
+		return
+	}
+	klog.Info("creates a loop device, data: %v", device)
+	if device.ImgDir == "" {
+		klog.Error("no device directory passed")
+		msg = "no device directory passed"
+		klog.Error(msg)
+		WrapResult(msg, UnprocessableEntityErrorCode, w)
+		return
+	} else if device.Size <= 0 {
+		klog.Error("no device size passed")
+		msg = "no device size passed"
+		klog.Error(msg)
+		WrapResult(msg, UnprocessableEntityErrorCode, w)
+		return
+	}
+
+	err := device.createDiskImage()
+	if err != nil {
+		msg = fmt.Sprintf("failed to create disk image, Error: %s", err.Error())
+		klog.Error(msg)
+		WrapResult(msg, ErrGeneral, w)
+		return
+	}
+
+	// create the loop device using losetup
+	// createLoopDeviceCommand := fmt.Sprintf("losetup -f %s --show", device.ImageName)
+	createLoopDeviceCommand := fmt.Sprintf("losetup -P /dev/%s %s --show", getLoopDevName(), device.ImageName)
```

Comment:
rather than getLoopDevName, just `losetup -f` I think would be a more robust approach.

Reply:
changed to `[losetup -f](losetup -f)`